{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#import the libraries\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import re\n",
    "import seaborn as sns\n",
    "from nltk.corpus import stopwords \n",
    "from nltk.stem.porter import PorterStemmer \n",
    "from sklearn.feature_extraction.text import CountVectorizer\n",
    "from sklearn.metrics import accuracy_score\n",
    "from sklearn import svm\n",
    "from sklearn import preprocessing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#load the dataset\n",
    "training_variants = pd.read_csv('training_variants.csv')\n",
    "training_text = pd.read_csv('training_text.csv',sep='\\|\\|', skiprows=1, engine='python',names=[\"ID\",\"text\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>CBL</td>\n",
       "      <td>V391I</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>CBL</td>\n",
       "      <td>V430M</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>7</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>8</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Y371H</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>9</td>\n",
       "      <td>CBL</td>\n",
       "      <td>C384R</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID    Gene             Variation  Class\n",
       "0   0  FAM58A  Truncating Mutations      1\n",
       "1   1     CBL                 W802*      2\n",
       "2   2     CBL                 Q249E      2\n",
       "3   3     CBL                 N454D      3\n",
       "4   4     CBL                 L399V      4\n",
       "5   5     CBL                 V391I      4\n",
       "6   6     CBL                 V430M      5\n",
       "7   7     CBL              Deletion      1\n",
       "8   8     CBL                 Y371H      4\n",
       "9   9     CBL                 C384R      4"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# EDA \n",
    "training_variants.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>7</td>\n",
       "      <td>CBL is a negative regulator of activated recep...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>8</td>\n",
       "      <td>Abstract Juvenile myelomonocytic leukemia (JM...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>9</td>\n",
       "      <td>Abstract Juvenile myelomonocytic leukemia (JM...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID                                               text\n",
       "0   0  Cyclin-dependent kinases (CDKs) regulate a var...\n",
       "1   1   Abstract Background  Non-small cell lung canc...\n",
       "2   2   Abstract Background  Non-small cell lung canc...\n",
       "3   3  Recent evidence has demonstrated that acquired...\n",
       "4   4  Oncogenic mutations in the monomeric Casitas B...\n",
       "5   5  Oncogenic mutations in the monomeric Casitas B...\n",
       "6   6  Oncogenic mutations in the monomeric Casitas B...\n",
       "7   7  CBL is a negative regulator of activated recep...\n",
       "8   8   Abstract Juvenile myelomonocytic leukemia (JM...\n",
       "9   9   Abstract Juvenile myelomonocytic leukemia (JM..."
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# EDA\n",
    "training_text.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<bound method NDFrame.describe of         ID    Gene             Variation  Class\n",
       "0        0  FAM58A  Truncating Mutations      1\n",
       "1        1     CBL                 W802*      2\n",
       "2        2     CBL                 Q249E      2\n",
       "3        3     CBL                 N454D      3\n",
       "4        4     CBL                 L399V      4\n",
       "5        5     CBL                 V391I      4\n",
       "6        6     CBL                 V430M      5\n",
       "7        7     CBL              Deletion      1\n",
       "8        8     CBL                 Y371H      4\n",
       "9        9     CBL                 C384R      4\n",
       "10      10     CBL                 P395A      4\n",
       "11      11     CBL                 K382E      4\n",
       "12      12     CBL                 R420Q      4\n",
       "13      13     CBL                 C381A      4\n",
       "14      14     CBL                 P428L      5\n",
       "15      15     CBL                 D390Y      4\n",
       "16      16     CBL  Truncating Mutations      1\n",
       "17      17     CBL                 Q367P      4\n",
       "18      18     CBL                 M374V      5\n",
       "19      19     CBL                 Y371S      4\n",
       "20      20     CBL                  H94Y      6\n",
       "21      21     CBL                 C396R      4\n",
       "22      22     CBL                 G375P      4\n",
       "23      23     CBL                 S376F      4\n",
       "24      24     CBL                 P417A      4\n",
       "25      25     CBL                 H398Y      4\n",
       "26      26   SHOC2                   S2G      4\n",
       "27      27    TERT                 Y846C      4\n",
       "28      28    TERT                 C228T      7\n",
       "29      29    TERT                 H412Y      4\n",
       "...    ...     ...                   ...    ...\n",
       "3291  3291     RET                 C620R      7\n",
       "3292  3292     RET                 C634Y      7\n",
       "3293  3293     RET                 V804G      1\n",
       "3294  3294     RET                 R886W      7\n",
       "3295  3295     RET                 F893L      4\n",
       "3296  3296     RET                 Y791F      7\n",
       "3297  3297   RUNX1                 R177*      4\n",
       "3298  3298   RUNX1                 Y113*      4\n",
       "3299  3299   RUNX1                 R139G      4\n",
       "3300  3300   RUNX1                  K83N      4\n",
       "3301  3301   RUNX1                 R177Q      4\n",
       "3302  3302   RUNX1                 R166Q      4\n",
       "3303  3303   RUNX1                 P173S      4\n",
       "3304  3304   RUNX1                 R201Q      4\n",
       "3305  3305   RUNX1              S70fsX93      4\n",
       "3306  3306   RUNX1                 W279*      1\n",
       "3307  3307   RUNX1  Truncating Mutations      1\n",
       "3308  3308   RUNX1                 R174*      7\n",
       "3309  3309   RUNX1                 D171G      1\n",
       "3310  3310   RUNX1         Amplification      7\n",
       "3311  3311   RUNX1     RUNX1-EVI1 Fusion      4\n",
       "3312  3312   RUNX1      TEL-RUNX1 Fusion      4\n",
       "3313  3313   RUNX1                  H78Q      4\n",
       "3314  3314   RUNX1                  G42R      6\n",
       "3315  3315   RUNX1  RUNX1-RUNX1T1 Fusion      4\n",
       "3316  3316   RUNX1                 D171N      4\n",
       "3317  3317   RUNX1                 A122*      1\n",
       "3318  3318   RUNX1               Fusions      1\n",
       "3319  3319   RUNX1                  R80C      4\n",
       "3320  3320   RUNX1                  K83E      4\n",
       "\n",
       "[3321 rows x 4 columns]>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# EDA\n",
    "training_variants.describe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<bound method NDFrame.describe of         ID                                               text\n",
       "0        0  Cyclin-dependent kinases (CDKs) regulate a var...\n",
       "1        1   Abstract Background  Non-small cell lung canc...\n",
       "2        2   Abstract Background  Non-small cell lung canc...\n",
       "3        3  Recent evidence has demonstrated that acquired...\n",
       "4        4  Oncogenic mutations in the monomeric Casitas B...\n",
       "5        5  Oncogenic mutations in the monomeric Casitas B...\n",
       "6        6  Oncogenic mutations in the monomeric Casitas B...\n",
       "7        7  CBL is a negative regulator of activated recep...\n",
       "8        8   Abstract Juvenile myelomonocytic leukemia (JM...\n",
       "9        9   Abstract Juvenile myelomonocytic leukemia (JM...\n",
       "10      10  Oncogenic mutations in the monomeric Casitas B...\n",
       "11      11  Noonan syndrome is an autosomal dominant conge...\n",
       "12      12  Noonan syndrome is an autosomal dominant conge...\n",
       "13      13  Noonan syndrome is an autosomal dominant conge...\n",
       "14      14  Oncogenic mutations in the monomeric Casitas B...\n",
       "15      15  Noonan syndrome is an autosomal dominant conge...\n",
       "16      16  To determine if residual cylindrical refractiv...\n",
       "17      17  Acquired uniparental disomy (aUPD) is a common...\n",
       "18      18  Oncogenic mutations in the monomeric Casitas B...\n",
       "19      19  Acquired uniparental disomy (aUPD) is a common...\n",
       "20      20   Abstract Background  Non-small cell lung canc...\n",
       "21      21  Oncogenic mutations in the monomeric Casitas B...\n",
       "22      22  Oncogenic mutations in the monomeric Casitas B...\n",
       "23      23  Recent evidence has demonstrated that acquired...\n",
       "24      24  Recent evidence has demonstrated that acquired...\n",
       "25      25  Recent evidence has demonstrated that acquired...\n",
       "26      26   Abstract N-myristoylation is a common form of...\n",
       "27      27  Heterozygous mutations in the telomerase compo...\n",
       "28      28  Sequencing studies have identified many recurr...\n",
       "29      29  Heterozygous mutations in the telomerase compo...\n",
       "...    ...                                                ...\n",
       "3291  3291  We investigatedthe transformingactivityofthe r...\n",
       "3292  3292  We investigatedthe transformingactivityofthe r...\n",
       "3293  3293  RET is a transmembrane tyrosine kinase partici...\n",
       "3294  3294  Introduction  Inherited germ line activating m...\n",
       "3295  3295  Many missense mutations in the RET proto-oncog...\n",
       "3296  3296  The RET proto-oncogene encodes a receptor tyro...\n",
       "3297  3297  THE AML1 GENE IS KNOWN as the most frequent ta...\n",
       "3298  3298  Introduction  Myelodysplastic syndromes (MDS) ...\n",
       "3299  3299  The BCR-ABL fusion protein generated by t(9;22...\n",
       "3300  3300  The most frequent mutations associated with le...\n",
       "3301  3301  The most frequent mutations associated with le...\n",
       "3302  3302  Familial platelet disorder with predisposition...\n",
       "3303  3303  Introduction  Myelodysplastic syndromes (MDS) ...\n",
       "3304  3304  Familial platelet disorder with predisposition...\n",
       "3305  3305  Introduction  Myelodysplastic syndromes (MDS) ...\n",
       "3306  3306  Here we report two new RUNX1 mutations in one ...\n",
       "3307  3307  The RUNX genes have come to prominence recentl...\n",
       "3308  3308   Familial platelet disorder with propensity to...\n",
       "3309  3309  The BCR-ABL fusion protein generated by t(9;22...\n",
       "3310  3310  RUNX proteins belong to a family of metazoan t...\n",
       "3311  3311  The AML1/EVI-1 chimeric gene is generated by t...\n",
       "3312  3312  Balanced chromosomal translocations are freque...\n",
       "3313  3313  The BCR-ABL fusion protein generated by t(9;22...\n",
       "3314  3314  Introduction  Myelodysplastic syndromes (MDS) ...\n",
       "3315  3315  The RUNX gene family includes three evolutiona...\n",
       "3316  3316  Introduction  Myelodysplastic syndromes (MDS) ...\n",
       "3317  3317  Introduction  Myelodysplastic syndromes (MDS) ...\n",
       "3318  3318  The Runt-related transcription factor 1 gene (...\n",
       "3319  3319  The RUNX1/AML1 gene is the most frequent targe...\n",
       "3320  3320  The most frequent mutations associated with le...\n",
       "\n",
       "[3321 rows x 2 columns]>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# EDA\n",
    "training_text.describe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3321, 4)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# EDA\n",
    "training_variants.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3321, 2)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# EDA\n",
    "training_text.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAtcAAAJaCAYAAADkoGtbAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJzt3X2clnWd9//3MMAQDIi62vXbFFc20MqLJAw1EQUt1DRR\nkWS68PZyk7xyYV1CFCHvQrNQ19abWr22UCQzlbQuLJHCwEWXX95k0hrrTSLhDWoMKgxw/v7w51yS\nQMP6nTln7Pl8PHw8Zo7zPDg/5+EMvOY7x3mcNZVKpRIAAOA961TtAQAA4P1CXAMAQCHiGgAAChHX\nAABQiLgGAIBCxDUAABQiroEOYY899shRRx2Vo48+uvm/8847r9pjtYnHHnssw4cPT5Lccsst+fa3\nv73V+//gBz/IzTffvNnb3rn/8OHD89hjj23TLL///e/z5S9/OUmycuXKnHDCCdu0/3/VtGnTMnz4\n8FxxxRUtuv/VV1+dCy+8sJWnAni3ztUeAKClvvvd72aHHXao9hhVNWbMmD97nyVLlqRfv37/5f23\n5vnnn89TTz2VJPngBz+Y2bNnv6c/r6W+//3v5+c//3n+23/7b23yeAD/VeIa6PD22muvHHLIIVm6\ndGm+8Y1vpHv37rnkkkvy6quvZsOGDRk7dmxGjRqVJLnqqqty1113Zfvtt88+++yTX//615k5c2bO\nOeec9OvXL6eddlqSbPL5ypUrc+GFF2bFihVpamrKZz/72Zxxxhl57rnncvLJJ+eggw7KI488ktde\ney0TJkzIEUcckfXr1+fyyy/Pz3/+89TW1mbgwIGZNm1ajjrqqJx//vkZMmRIkmTKlCnp169fTjrp\npE2e06xZs/Ld73439fX16d+/f/P2q6++Oq+88kqmTp2aWbNmZfbs2enSpUvq6upy4YUX5qmnnsp9\n992XhQsXplu3blm1alUefvjhvPDCC9ljjz2y2267Ne//9uMsXbo069atyymnnJJRo0Zl8eLFueii\ni3L33XcnSfPnc+bMyZQpU7Jy5cqcdtppueCCC3LUUUflV7/6VZqamnLppZfmgQceSG1tbQYMGJDJ\nkyenvr4+w4cPzzHHHJMHHnggK1asyOGHH56vfOUr7/r/+OSTT+bCCy/Mq6++mpqampx66qkZOXJk\nGhoaUqlUcvrpp2fatGnZZ599mvfZ0nF+p/nz5+f666/PunXrsmrVqowcOTLjx4/PmjVrMnny5Dzz\nzDPp1KlTPvaxj+XCCy/MG2+8sdntnTp1yn333Zdrr702TU1N6datWyZNmpSBAwdm2bJlOe+887Ju\n3bpUKpWMGjUqX/jCF97rlzbQAYlroMM46aST0qnT/z2b7cYbb8yOO+6YpqamDBs2LFdddVXWr1+f\no48+Ol//+tfzsY99LKtXr87nP//5fPjDH84LL7yQn/70p7nzzjtTV1eXL33pSy163IkTJ+bkk0/O\n8OHDs3bt2px++unp06dPBgwYkN///vcZMmRIzj///Nxzzz259NJLc8QRR2TWrFl5/PHHM2fOnHTt\n2jX/8A//kJ/85CcZM2ZMfvCDH2TIkCFpbGzMvHnz3hWaTzzxRL71rW9lzpw52WmnnZpD+J02bNiQ\nr33ta7nvvvuy8847584778ySJUvy+c9/PvPmzUu/fv3yhS98IVdffXWWL1+eu+++O507d87VV1+9\nyZ9TV1eXO+64IytXrszIkSPz8Y9/fIvHoba2NhdffHEuuuii3HDDDXnuueeab7v22mvzwgsvZM6c\nOamtrc15552Xr3/9682nZrz++uuZNWtWVq5cmU9/+tMZM2ZMdt111+b9169fn3HjxuUrX/lKPvOZ\nz2TlypU5/vjjs9tuu2XWrFnZY489Nvubiy0d57dVKpXceOONufTSS/M3f/M3WblyZYYNG5YTTzwx\nCxYsyJo1azJnzpxs2LAh06ZNy+9///v86le/2uz2SqWSK664It/73vey/fbb58knn8wpp5ySn/70\np7nhhhsyfPjw/N3f/V1efPHFfO1rX8uYMWM2+XoF/jKIa6DD2NppIW+vZj799NN59tlnc+655zbf\n9uabb+Y3v/lNfve73+XTn/506uvrkySf//zn893vfnerj/n666/noYceymuvvZarrrqqedvSpUsz\nYMCAdOnSJQcddFCS5KMf/WheffXVJMmiRYty9NFHp1u3bkmSK6+8Mknyxz/+Mf/8z/+cVatWZe7c\nuTn44IPTq1evTR7zgQceyAEHHJCddtqpec5f/vKXm9yntrY2hx12WE444YQcfPDBOeCAA3LUUUdt\n9jnsvffe6dx583/dv33O9Ac/+MEMGTIkDzzwQPbYY4+tHpPNWbBgQSZMmJAuXbokScaOHZszzzyz\n+fZDDjmk+XF23HHHvPbaa5vE9dNPP521a9fmM5/5TPP9PvOZz+T+++/PwIEDt/i4WzrOb/8QUVNT\nk+uuuy4///nPc/fdd2fZsmWpVCp54403MmjQoFxxxRUZO3ZsPvWpT+Wkk07Kbrvtlk6dOm12+803\n35wXXnghJ598cvPj19TU5Nlnn82nP/3pTJo0KY8++mj233//TJkyRVjDXyhxDbwvdO/ePclbK7q9\nevXKnDlzmm976aWX0rNnz1x55ZWpVCrN298OweStSHrnbU1NTUmSjRs3plKpZPbs2fnABz6QJFm1\nalXq6uryyiuvpEuXLs0RVVNT07z/n8bsSy+9lI0bN2bnnXfOYYcdlh/96Ee566673nUKw+Zmqa2t\n3exz/sY3vpH/+I//yKJFi/Kd73wnt912W6699totHpvNeWcAViqVdO7ceYvHYms2btz4rs/fuV9d\nXV3zx3/6529u/7fnWb9+/VYfd0vH+W2vv/56jjnmmBx66KHZZ599ctxxx+Xee+9NpVLJrrvump/9\n7GdZvHhx/u3f/i2nnHJKpkyZksMOO2yz2zdu3Jj999+/OeCTZMWKFdl5552z55575p577smiRYvy\nwAMP5J//+Z8ze/bs9OnTZ6vzA+8/fqwG3ld233331NXVNcf1ihUrcuSRR+bXv/51Dj744MydOzev\nvfZaNm7cmDvvvLN5v+233z6//vWvk7wVz//+7/+eJKmvr8/ee++d//2//3eSt1aex4wZk3nz5m11\njv333z9333131q1bl40bN+arX/1qfvzjHydJvvCFL+R73/teKpVKBgwY8K59P/WpT2XhwoX5wx/+\nkCS544473nWfVatW5aCDDkrv3r1z8sknZ/z48fntb3+b5K0Y/3NR+ra3/+znn38+ixYtyv77758d\ndtghzz//fF5++eVUKpXce++9zfevra3dbGwfeOCBmT17dpqamrJx48bcfPPNOeCAA1o0Q/LW/7cu\nXbrkpz/9aZK3rkRyzz335FOf+tRW99vacU6SZ555Jo2NjRk/fnyGDx+eBx98sPm+s2bNyuTJkzNk\nyJBMnDgxQ4YMyZNPPrnF7fvtt18WLlyYZcuWJUl+8Ytf5HOf+1zWrl2bs88+Oz/5yU/y2c9+NtOm\nTUt9fX1WrFjR4ucPvH9YuQbeV7p27Zprrrkml1xySf7lX/4l69evz9///d9n0KBBSZITTzwxDQ0N\nqaury4c+9KHm/caOHZt//Md/zIgRI7LLLrtk8ODBzbd94xvfyEUXXZSjjjoq69aty5FHHpnPfe5z\nm5xz/KdOOOGELF++PMcee2wqlUoGDx6csWPHJkn23HPPbLfddlu8jN0ee+yRiRMn5qSTTkqPHj02\nG+A77LBDxo0bl5NPPjndunVrPh86SYYOHZqLLrqoRcdr7dq1OeaYY9LU1JQpU6Zk9913b57/uOOO\ny0477ZSDDz64+f79+vVLbW1tRo0atcll8caNG5fLLrssI0eOzPr16zNgwICcf/75LZoheeu3CNdc\nc00uvvjiXH311dmwYUPOPPPM7Lffflvdb0vH+e0V/D322CMHH3xwDj/88PTq1St9+vTJhz/84Tzz\nzDMZOXJkHnzwwRxxxBH5wAc+kL/+67/OiSeemC5dumx2+3bbbZcLL7ww//AP/9C8yn/ttdeme/fu\n+dKXvpTzzjsv3//+91NbW5tDDz10k68h4C9HTeVPfzcH8Bdi7ty5ufnmmzNz5sw2fdxnn302Y8eO\nzdy5c5tPNQHg/cFpIQBt6KqrrsqYMWMyadIkYQ3wPmTlGgAACmnVletHHnmk+RzDZ555JmPGjElD\nQ0OmTZvW/GruW2+9Nccee2xGjx6d+fPnJ3nrsllf/vKX09DQkNNPPz2rVq1qzTEBAKCIVovr73zn\nO5kyZUrWrl2bJJk+fXrGjx+fWbNmpVKpZN68eXnxxRczc+bMzJ49OzfccENmzJiRdevW5ZZbbkn/\n/v0za9asjBw5Mtdcc01rjQkAAMW0Wlz36dNnk3cCe/zxx5tfOT106NAsWrQojz76aAYOHJiuXbum\nZ8+e6dOnT5YuXZolS5bkwAMPbL7vAw880FpjAgBAMa12Kb4RI0ZscpmqSqXS/AYLPXr0yOrVq9PY\n2JiePXs236dHjx5pbGzcZPvb922JJUuWFHwGAACweW9f4vVPtdnVQt75LmBr1qxJr169Ul9fnzVr\n1myyvWfPnptsf/u+AADQ3rXZm8h89KMfzeLFi7PvvvtmwYIF2W+//TJgwIBceeWVWbt2bdatW5dl\ny5alf//++cQnPpFf/OIXGTBgQBYsWLDFnww2Z1vuCwAA22prZ0u0WVxPmjQp559/fmbMmJG+fftm\nxIgRqa2tzdixY9PQ0JBKpZIJEyakrq6u+RqwY8aMSZcuXfLNb36zrcYEAID/svfVda6XLFli5RoA\ngFa1teb0Do0AAFCIuAYAgELENQAAFCKuAQCgEHENAACFiGsAAChEXAMAQCHiGgAAChHXAABQiLgG\nAIBCxDUAABQirgEAoBBxDQAAhYhrAAAoRFwDAEAh4hoAAAoR1wAAUIi4BgCAQsQ1AAAUIq4BAKAQ\ncQ0AAIV0rvYAAEDr+OCN36r2CG1m5an/q9ojQBIr1wAAUIy4BgCAQsQ1AAAUIq4BAKAQcQ0AAIWI\nawAAKERcAwBAIeIaAAAKEdcAAFCIuAYAgELENQAAFCKuAQCgEHENAACFiGsAAChEXAMAQCHiGgAA\nChHXAABQiLgGAIBCxDUAABQirgEAoBBxDQAAhYhrAAAoRFwDAEAh4hoAAAoR1wAAUIi4BgCAQsQ1\nAAAUIq4BAKAQcQ0AAIWIawAAKERcAwBAIeIaAAAKEdcAAFCIuAYAgELENQAAFCKuAQCgEHENAACF\niGsAAChEXAMAQCHiGgAAChHXAABQiLgGAIBCxDUAABQirgEAoBBxDQAAhYhrAAAoRFwDAEAh4hoA\nAAoR1wAAUIi4BgCAQsQ1AAAUIq4BAKAQcQ0AAIWIawAAKERcAwBAIeIaAAAKEdcAAFCIuAYAgELE\nNQAAFCKuAQCgEHENAACFiGsAAChEXAMAQCHiGgAAChHXAABQiLgGAIBCxDUAABQirgEAoBBxDQAA\nhYhrAAAoRFwDAEAh4hoAAAoR1wAAUIi4BgCAQsQ1AAAUIq4BAKAQcQ0AAIWIawAAKERcAwBAIZ3b\n8sGamppyzjnnZPny5enUqVMuuuiidO7cOeecc05qamrSr1+/TJs2LZ06dcqtt96a2bNnp3Pnzhk3\nblyGDRvWlqMCAMA2a9O4/sUvfpH169dn9uzZWbhwYa688so0NTVl/Pjx2XfffTN16tTMmzcve++9\nd2bOnJkf/vCHWbt2bRoaGnLAAQeka9eubTkuAABskzY9LWT33XfPhg0bsnHjxjQ2NqZz5855/PHH\nM3jw4CTJ0KFDs2jRojz66KMZOHBgunbtmp49e6ZPnz5ZunRpW44KAADbrE1Xrrt3757ly5fn8MMP\nzyuvvJLrrrsuDz30UGpqapIkPXr0yOrVq9PY2JiePXs279ejR480Nja26DGWLFnSKrMDAO2Xf/9p\nL9o0rv/1X/81Q4YMydlnn50VK1bkpJNOSlNTU/Pta9asSa9evVJfX581a9Zssv2dsb01gwYNKj43\nAHRIjzxQ7QnajH//aUtb+2GuTU8L6dWrV3Mkb7fddlm/fn0++tGPZvHixUmSBQsWZJ999smAAQOy\nZMmSrF27NqtXr86yZcvSv3//thwVAAC2WZuuXJ988sk599xz09DQkKampkyYMCF77bVXzj///MyY\nMSN9+/bNiBEjUltbm7Fjx6ahoSGVSiUTJkxIXV1dW44KAADbrKZSqVSqPUQpS5Ys8WshAPj/ffDG\nb1V7hDaz8tT/Ve0R+Auyteb0JjIAAFCIuAYAgELENQAAFCKuAQCgEHENAACFiGsAAChEXAMAQCHi\nGgAAChHXAABQiLgGAIBCxDUAABQirgEAoBBxDQAAhYhrAAAoRFwDAEAh4hoAAAoR1wAAUIi4BgCA\nQsQ1AAAUIq4BAKAQcQ0AAIWIawAAKERcAwBAIeIaAAAKEdcAAFCIuAYAgELENQAAFCKuAQCgEHEN\nAACFiGsAAChEXAMAQCHiGgAAChHXAABQiLgGAIBCxDUAABQirgEAoBBxDQAAhYhrAAAoRFwDAEAh\n4hoAAAoR1wAAUIi4BgCAQsQ1AAAUIq4BAKAQcQ0AAIWIawAAKERcAwBAIeIaAAAKEdcAAFCIuAYA\ngELENQAAFCKuAQCgEHENAACFiGsAAChEXAMAQCHiGgAAChHXAABQSOdqDwC8v/31F++u9ght6vnr\nj6z2CABUkZVrAAAoRFwDAEAh4hoAAAoR1wAAUIi4BgCAQsQ1AAAUIq4BAKAQcQ0AAIWIawAAKERc\nAwBAIeIaAAAKEdcAAFCIuAYAgELENQAAFCKuAQCgEHENAACFiGsAAChEXAMAQCHiGgAAChHXAABQ\niLgGAIBCxDUAABQirgEAoBBxDQAAhYhrAAAoRFwDAEAh4hoAAAoR1wAAUIi4BgCAQsQ1AAAUIq4B\nAKAQcQ0AAIWIawAAKERcAwBAIeIaAAAKEdcAAFCIuAYAgELENQAAFCKuAQCgEHENAACFiGsAAChE\nXAMAQCGd2/oBr7/++tx3331pamrKmDFjMnjw4JxzzjmpqalJv379Mm3atHTq1Cm33nprZs+enc6d\nO2fcuHEZNmxYW48KAADbpE1XrhcvXpxf/epXueWWWzJz5sz84Q9/yPTp0zN+/PjMmjUrlUol8+bN\ny4svvpiZM2dm9uzZueGGGzJjxoysW7euLUcFAIBt1qZx/ctf/jL9+/fPmWeemTPOOCMHH3xwHn/8\n8QwePDhJMnTo0CxatCiPPvpoBg4cmK5du6Znz57p06dPli5d2pajAgDANmvT00JeeeWVPP/887nu\nuuvy3HPPZdy4calUKqmpqUmS9OjRI6tXr05jY2N69uzZvF+PHj3S2NjYlqMCAMA2a9O47t27d/r2\n7ZuuXbumb9++qauryx/+8Ifm29esWZNevXqlvr4+a9as2WT7O2N7a5YsWVJ8boCW8ncQVIfvPdqL\nNo3rQYMG5Xvf+15OOeWUvPDCC3njjTey//77Z/Hixdl3332zYMGC7LfffhkwYECuvPLKrF27NuvW\nrcuyZcvSv3//Fj8G0J7cXe0B2pS/g2hXHnmg2hO0Gd97tKWt/TDXpnE9bNiwPPTQQxk1alQqlUqm\nTp2aXXbZJeeff35mzJiRvn37ZsSIEamtrc3YsWPT0NCQSqWSCRMmpK6uri1HBQCAbdbml+L7yle+\n8q5tN91007u2jR49OqNHj26LkQAAoAhvIgMAAIWIawAAKERcAwBAIeIaAAAKEdcAAFCIuAYAgELE\nNQAAFCKuAQCgEHENAACFiGsAAChEXAMAQCHiGgAAChHXAABQSOdqD9BWdvvE49UeoU098/9+rNoj\nAAD8xbFyDQAAhYhrAAAoRFwDAEAh4hoAAAoR1wAAUIi4BgCAQsQ1AAAUIq4BAKAQcQ0AAIWIawAA\nKERcAwBAIeIaAAAKaVFcn3766fk//+f/pKmpqbXnAQCADqtFcf13f/d3uf/++zNixIhccMEFefTR\nR1t7LgAA6HA6t+ROn/zkJ/PJT34yb775ZubOnZuzzjor9fX1GTVqVBoaGtK1a9fWnhMAANq9FsV1\nkixevDhz5szJwoULM3To0BxxxBFZuHBhxo0blxtuuKE1ZwQAgA6hRXE9bNiw7LLLLjnuuOMyderU\ndOvWLUkyePDgjBo1qlUHBACAjqJFcf3d7343PXr0yI477pg333wzzzzzTHbbbbfU1tbmjjvuaO0Z\nAQCgQ2jRCxp//vOf53/+z/+ZJHn55Zdzxhln5Pvf/36rDgYAAB1Ni+L61ltvzc0335wk+dCHPpTb\nb789N910U6sOBgAAHU2L4rqpqWmTK4J06dKl1QYCAICOqkXnXB966KE56aSTcvjhhydJfvrTn2b4\n8OGtOhgAAHQ0LYrriRMnZu7cuXnooYfSuXPnnHjiiTn00ENbezYAAOhQWnyd67/927/NX/3VX6VS\nqSRJHnrooXzyk59stcEAAKCjaVFcX3DBBZk/f3523XXX5m01NTX53ve+12qDAQBAR9OiuF64cGHm\nzp3b/OYxAADAu7XoaiG77rpr8+kgAADA5rVo5Xq77bbLZz/72QwcOHCTS/JNnz691QYDAICOpkVx\nfeCBB+bAAw9s7VkAAKBDa1FcH3PMMXnuuefyu9/9LkOGDMmKFSs2eXEjAADQwnOuf/KTn2TcuHG5\n5JJL8tprr+WEE07InDlzWns2AADoUFoU19/5zndyyy23pEePHtlxxx1zxx135Nvf/nZrzwYAAB1K\ni+K6U6dOqa+vb/585513TqdOLdoVAAD+YrTonOt+/frlpptuyvr16/PEE09k1qxZ2XPPPVt7NgAA\n6FBatPw8derUrFy5MnV1dTn33HNTX1+fadOmtfZsAADQobRo5bp79+45++yzc/bZZ7f2PAAA0GG1\nKK733HPP1NTUbLJtp512yoIFC1plKAAA6IhaFNdLly5t/ripqSn33ntvHn744VYbCgAAOqJtvuRH\nly5dcvjhh+ff/u3fWmMeAADosFq0cn3nnXc2f1ypVPLkk0+mS5curTYUAAB0RC2K68WLF2/y+fbb\nb58rrriiVQYCAICOqkVxPX369NaeAwAAOrwWxfXw4cPfdbWQ5K1TRGpqajJv3rzigwEAQEfTorg+\n6qij0qVLl4wePTqdO3fOXXfdlcceeywTJkxo7fkAAKDDaFFc33///bn99tubPz/ppJNy7LHH5kMf\n+lCrDQYAAB1Niy/Ft2jRouaP58+fnx49erTKQAAA0FG1aOX6wgsvzKRJk/LSSy8lSfr27ZvLLrus\nVQcDAICOpkVxvddee+XHP/5xVq1albq6OqvWAACwGS06LWT58uU55ZRTcsIJJ+T111/PiSeemOee\ne661ZwMAgA6lRXE9derUnHbaaenevXv+6q/+KkceeWQmTZrU2rMBAECH0qK4fuWVVzJkyJAkSU1N\nTUaPHp3GxsZWHQwAADqaFsV1t27d8oc//KH5jWT+/d//PV27dm3VwQAAoKNp0QsaJ0+enC9+8Yt5\n9tlnc/TRR+e1117LVVdd1dqzAQBAh9KiuH755Zdz22235emnn86GDRvSt29fK9cAAPAnWnRayOWX\nX54uXbqkX79+2XPPPYU1AABsRotWrnfddddMnjw5H//4x9OtW7fm7SNHjmy1wQAAoKPZalyvXLky\nH/zgB7P99tsnSR555JFNbhfXAADwf201rs8444zccccdmT59em688caceuqpbTUXAAB0OFs957pS\nqTR/fNddd7X6MAAA0JFtNa7fvq51smloAwAA79aiq4Ukm4Y2AADwbls95/rJJ5/MIYcckuStFze+\n/XGlUklNTU3mzZvX+hMCAEAHsdW4vueee9pqDtqRPoc+WO0R2syz9w6u9ggAwPvIVuP6Qx/6UFvN\nAQAAHV6Lz7kGAAC2TlwDAEAh4hoAAAoR1wAAUIi4BgCAQsQ1AAAUIq4BAKAQcQ0AAIWIawAAKERc\nAwBAIeIaAAAKEdcAAFCIuAYAgELENQAAFCKuAQCgEHENAACFiGsAAChEXAMAQCHiGgAAChHXAABQ\niLgGAIBCxDUAABQirgEAoJCqxPXLL7+cgw46KMuWLcszzzyTMWPGpKGhIdOmTcvGjRuTJLfeemuO\nPfbYjB49OvPnz6/GmAAAsE3aPK6bmpoyderUdOvWLUkyffr0jB8/PrNmzUqlUsm8efPy4osvZubM\nmZk9e3ZuuOGGzJgxI+vWrWvrUQEAYJu0eVxfdtllOeGEE7LzzjsnSR5//PEMHjw4STJ06NAsWrQo\njz76aAYOHJiuXbumZ8+e6dOnT5YuXdrWowIAwDZp07i+/fbbs8MOO+TAAw9s3lapVFJTU5Mk6dGj\nR1avXp3Gxsb07Nmz+T49evRIY2NjW44KAADbrHNbPtgPf/jD1NTU5IEHHsgTTzyRSZMmZdWqVc23\nr1mzJr169Up9fX3WrFmzyfZ3xvbWLFmyZAu3dHsvo3c4Wz4OvJPjRGm+pqA6fO/RXrRpXN98883N\nH48dOzZf/epXc/nll2fx4sXZd999s2DBguy3334ZMGBArrzyyqxduzbr1q3LsmXL0r9//xY9xqBB\ng7Zwy+MFnkHHseXj0BIPFpujvXtvx4mWubvaA7QpX1O0K488UO0J2ozvPdrS1n6Ya9O43pxJkybl\n/PPPz4wZM9K3b9+MGDEitbW1GTt2bBoaGlKpVDJhwoTU1dVVe1QAANiqqsX1zJkzmz++6aab3nX7\n6NGjM3r06LYcCQAA3hNvIgMAAIWIawAAKERcAwBAIeIaAAAKEdcAAFCIuAYAgELENQAAFCKuAQCg\nEHENAACFiGsAAChEXAMAQCHiGgAAChHXAABQiLgGAIBCxDUAABQirgEAoBBxDQAAhYhrAAAoRFwD\nAEAh4hoAAAoR1wAAUIi4BgCAQsQ1AAAUIq4BAKAQcQ0AAIWIawAAKERcAwBAIeIaAAAKEdcAAFCI\nuAYAgELENQAAFCKuAQCgEHENAACFiGsAAChEXAMAQCHiGgAAChHXAABQiLgGAIBCxDUAABQirgEA\noBBxDQAWj5WQAAAPKklEQVQAhYhrAAAoRFwDAEAh4hoAAAoR1wAAUIi4BgCAQsQ1AAAUIq4BAKAQ\ncQ0AAIWIawAAKERcAwBAIeIaAAAKEdcAAFCIuAYAgELENQAAFCKuAQCgEHENAACFiGsAAChEXAMA\nQCHiGgAAChHXAABQiLgGAIBCxDUAABTSudoDAMC2+ODXb6j2CG1q5VdOq/YIwDawcg0AAIWIawAA\nKERcAwBAIeIaAAAKEdcAAFCIuAYAgELENQAAFCKuAQCgEHENAACFiGsAAChEXAMAQCHiGgAAChHX\nAABQiLgGAIBCxDUAABQirgEAoBBxDQAAhXSu9gAAvOX/mTy72iO0mRXTT6j2CACtwso1AAAUIq4B\nAKAQcQ0AAIWIawAAKERcAwBAIeIaAAAKEdcAAFCIuAYAgELENQAAFCKuAQCgEHENAACFiGsAAChE\nXAMAQCHiGgAAChHXAABQiLgGAIBCxDUAABQirgEAoBBxDQAAhXRuywdramrKueeem+XLl2fdunUZ\nN25cPvzhD+ecc85JTU1N+vXrl2nTpqVTp0659dZbM3v27HTu3Dnjxo3LsGHD2nJUAADYZm0a1z/6\n0Y/Su3fvXH755Xn11VczcuTI7Lnnnhk/fnz23XffTJ06NfPmzcvee++dmTNn5oc//GHWrl2bhoaG\nHHDAAenatWtbjgsAANukTeP6sMMOy4gRI5IklUoltbW1efzxxzN48OAkydChQ7Nw4cJ06tQpAwcO\nTNeuXdO1a9f06dMnS5cuzYABA9pyXAAA2CZtGtc9evRIkjQ2Nuass87K+PHjc9lll6Wmpqb59tWr\nV6exsTE9e/bcZL/GxsYWPcaSJUu2cEu39zR7R7Pl48A7OU6U5muqZRynlnOsWsZxor1o07hOkhUr\nVuTMM89MQ0NDjjrqqFx++eXNt61Zsya9evVKfX191qxZs8n2d8b21gwaNGgLtzz+XsbucLZ8HFri\nwWJztHfv7TjRMndXe4A29Z6+pm57stwg7dx7Ok7zHi43SAfwno7VIw+UG6Sd8/c5bWlrP8y16dVC\nXnrppZx66qmZOHFiRo0alST56Ec/msWLFydJFixYkH322ScDBgzIkiVLsnbt2qxevTrLli1L//79\n23JUAADYZm26cn3dddflj3/8Y6655ppcc801SZLzzjsvF198cWbMmJG+fftmxIgRqa2tzdixY9PQ\n0JBKpZIJEyakrq6uLUcFAIBt1qZxPWXKlEyZMuVd22+66aZ3bRs9enRGjx7dFmMBAEAR3kQGAAAK\nEdcAAFCIuAYAgELENQAAFCKuAQCgEHENAACFiGsAAChEXAMAQCHiGgAAChHXAABQiLgGAIBCxDUA\nABQirgEAoBBxDQAAhYhrAAAoRFwDAEAh4hoAAAoR1wAAUIi4BgCAQsQ1AAAUIq4BAKAQcQ0AAIWI\nawAAKERcAwBAIeIaAAAKEdcAAFCIuAYAgELENQAAFCKuAQCgEHENAACFiGsAAChEXAMAQCHiGgAA\nChHXAABQiLgGAIBCxDUAABQirgEAoBBxDQAAhYhrAAAoRFwDAEAh4hoAAAoR1wAAUIi4BgCAQsQ1\nAAAUIq4BAKAQcQ0AAIWIawAAKERcAwBAIeIaAAAKEdcAAFCIuAYAgELENQAAFCKuAQCgEHENAACF\niGsAACikc7UHgI5ql+PnV3uENvPcD4ZVewQA6BCsXAMAQCHiGgAAChHXAABQiLgGAIBCxDUAABQi\nrgEAoBBxDQAAhYhrAAAoRFwDAEAh4hoAAAoR1wAAUIi4BgCAQsQ1AAAUIq4BAKAQcQ0AAIWIawAA\nKERcAwBAIeIaAAAKEdcAAFCIuAYAgELENQAAFCKuAQCgEHENAACFiGsAAChEXAMAQCHiGgAAChHX\nAABQiLgGAIBCOld7AAAA2r8+D06v9ght6tnBk/9L+1m5BgCAQsQ1AAAUIq4BAKAQcQ0AAIWIawAA\nKERcAwBAIeIaAAAKEdcAAFCIN5EBAP6i/fXd36z2CG3m+SPPrvYI73tWrgEAoBBxDQAAhYhrAAAo\nRFwDAEAh4hoAAAoR1wAAUEi7vRTfxo0b89WvfjW//e1v07Vr11x88cXZbbfdqj0WAABsUbtdub73\n3nuzbt26fP/738/ZZ5+dSy+9tNojAQDAVrXbuF6yZEkOPPDAJMnee++dX//611WeCAAAtq7dxnVj\nY2Pq6+ubP6+trc369eurOBEAAGxdTaVSqVR7iM2ZPn16Pv7xj+eII45IkgwdOjQLFizY6j5Llixp\ni9EAAPgLN2jQoM1ub7cvaPzEJz6R+fPn54gjjsjDDz+c/v37/9l9tvQkAQCgLbTbleu3rxbyH//x\nH6lUKvna176Wv/3bv632WAAAsEXtNq4BAKCjabcvaAQAgI5GXAMAQCHiGgAACmm3Vwt5P3jkkUfy\njW98IzNnzqz2KO1WU1NTzj333Cxfvjzr1q3LuHHjcsghh1R7rHZnw4YNmTJlSp566qnU1NTkggsu\naNEVdP5Svfzyyzn22GNz4403eiH0VhxzzDHN7yewyy67ZPr06VWeqH26/vrrc99996WpqSljxozJ\n8ccfX+2R2qXbb789d9xxR5Jk7dq1eeKJJ7Jw4cL06tWrypO1L01NTTnnnHOyfPnydOrUKRdddJG/\np7Zg3bp1mTx5cn7/+9+nvr4+U6dOzd/8zd9Ue6w/S1y3ku985zv50Y9+lA984APVHqVd+9GPfpTe\nvXvn8ssvz6uvvpqRI0eK682YP39+kmT27NlZvHhxrrjiilx77bVVnqp9ampqytSpU9OtW7dqj9Ku\nrV27NpVKxQ//f8bixYvzq1/9KrfcckveeOON3HjjjdUeqd069thjc+yxxyZJLrjgghx33HHCejN+\n8YtfZP369Zk9e3YWLlyYK6+8MldffXW1x2qXbr311nTv3j233npr/vM//zMXXXRRbrjhhmqP9Wc5\nLaSV9OnTxzdLCxx22GH5+7//+yRJpVJJbW1tlSdqnw499NBcdNFFSZLnn3/eP1hbcdlll+WEE07I\nzjvvXO1R2rWlS5fmjTfeyKmnnpoTTzwxDz/8cLVHapd++ctfpn///jnzzDNzxhln5OCDD672SO3e\nY489lt/97nf5/Oc/X+1R2qXdd989GzZsyMaNG9PY2JjOna1zbsnvfve7DB06NEnSt2/fLFu2rMoT\ntYz/o61kxIgRee6556o9RrvXo0ePJG+93f1ZZ52V8ePHV3mi9qtz586ZNGlSfvazn+Wf/umfqj1O\nu3T77bdnhx12yIEHHphvf/vb1R6nXevWrVtOO+20HH/88Xn66adz+umnZ+7cuf6h/xOvvPJKnn/+\n+Vx33XV57rnnMm7cuMydOzc1NTXVHq3duv7663PmmWdWe4x2q3v37lm+fHkOP/zwvPLKK7nuuuuq\nPVK79ZGPfCTz58/PoYcemkceeSQrV67Mhg0b2v1CnJVrqm7FihU58cQTc/TRR+eoo46q9jjt2mWX\nXZZ77rkn559/fl5//fVqj9Pu/PCHP8yiRYsyduzYPPHEE5k0aVJefPHFao/VLu2+++753Oc+l5qa\nmuy+++7p3bu3Y7UZvXv3zpAhQ9K1a9f07ds3dXV1WbVqVbXHarf++Mc/5qmnnsp+++1X7VHarX/9\n13/NkCFDcs8992TOnDk555xzsnbt2mqP1S4dd9xxqa+vT0NDQ372s5/lYx/7WLsP60RcU2UvvfRS\nTj311EycODGjRo2q9jjt1p133pnrr78+SfKBD3wgNTU16dTJt++fuvnmm3PTTTdl5syZ+chHPpLL\nLrssO+20U7XHapduu+22XHrppUmSlStXprGx0bHajEGDBuX+++9PpVLJypUr88Ybb6R3797VHqvd\neuihh7L//vtXe4x2rVevXunZs2eSZLvttsv69euzYcOGKk/VPj322GPZf//9c8stt+Swww7Lrrvu\nWu2RWsTv/6iq6667Ln/84x9zzTXX5Jprrkny1otBvRhtU5/5zGcyefLkfOELX8j69etz7rnnOka8\nJ6NGjcrkyZMzZsyY1NTU5Gtf+5pTQjZj2LBheeihhzJq1KhUKpVMnTq1Q6ycVctTTz2VXXbZpdpj\ntGsnn3xyzj333DQ0NKSpqSkTJkxI9+7dqz1Wu7TbbrvlqquuynXXXZeePXvmkksuqfZILeLtzwEA\noBC/VwYAgELENQAAFCKuAQCgEHENAACFiGsAACjEdZcA3kcaGxvzzW9+Mw899FBqa2vTq1evnHPO\nOWlsbMy3vvWtzJw5s9ojAryvWbkGeJ/YuHFjTj/99Gy33Xa58847M2fOnJx55pk5/fTT8+qrr1Z7\nPIC/CFauAd4nFi9enBdeeCFnnXVW8zt47rfffpk+fXrWrFnTfL8HH3wwV1xxRd5888289tprmThx\nYg4//PDcdddd+Zd/+ZfU1tZml112yeWXX55XXnkl//iP/5jXX389nTp1ypQpU7L33ntX6ykCtHtW\nrgHeJ37zm9/kv//3/94c1m876KCDsuOOOzZ/ftNNN+Xiiy/OHXfckUsuuaT53VGvvPLK3Hjjjbn9\n9tuz++675z//8z9z22235eCDD87tt9+eiRMnZsmSJW36nAA6GivXAO8TnTp1SkvedPfyyy/P/Pnz\nM3fu3DzyyCPNq9rDhg3LmDFjcsghh2TEiBH5yEc+ktdffz1f/vKX88QTT+Sggw7K//gf/6O1nwZA\nh2blGuB9Yq+99spvfvObdwX2jBkzNtnW0NCQRx99NHvttVfOOOOM5u1TpkzJP/3TP6V3796ZOHFi\n5syZk0GDBuXHP/5xhgwZkp/85Ceb3B+Ad7NyDfA+sc8++2THHXfMt771rXzpS19KbW1t7r///tx+\n++3Zc889kySvvvpqnn766cyaNSt1dXW5+uqrs2HDhqxfvz5HHHFEZs6cmS9+8YtpamrKE088kd/+\n9rfZeeedc/LJJ2fffffNMcccU+VnCdC+iWuA94mamppcc801mT59eo488sh07tw522+/fb797W9n\n9erVSZLevXvn+OOPz2c/+9nU19dn7733zptvvpl169blrLPOyimnnJJu3bqlV69eueyyy7Jx48ac\nffbZueOOO1JbW5tp06ZV+VkCtG81lZacoAcAAPxZzrkGAIBCxDUAABQirgEAoBBxDQAAhYhrAAAo\nRFwDAEAh4hoAAAoR1wAAUMj/B7JZAhgIY+vmAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0xae99550>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# quick visualization of frequency of classes 1-9\n",
    "sns.set(style=\"whitegrid\", color_codes=True)\n",
    "\n",
    "plt.figure(figsize=(12,10))\n",
    "ax = sns.countplot(x=\"Class\", data=training_variants,palette=\"winter\")\n",
    "plt.ylabel('Frequency')\n",
    "plt.xlabel('Class')\n",
    "plt.title('Frequency distribution of classes')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Find out unique Genes\n",
    "unique_genes = list(training_variants.Gene.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['FAM58A',\n",
       " 'CBL',\n",
       " 'SHOC2',\n",
       " 'TERT',\n",
       " 'DICER1',\n",
       " 'PTPRT',\n",
       " 'RHEB',\n",
       " 'SHQ1',\n",
       " 'CCND2',\n",
       " 'RAD50']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_genes[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "264"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#There are 264 unique genes\n",
    "len(unique_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Find out unique variation\n",
    "unique_variation = list(training_variants.Variation.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Truncating Mutations',\n",
       " 'W802*',\n",
       " 'Q249E',\n",
       " 'N454D',\n",
       " 'L399V',\n",
       " 'V391I',\n",
       " 'V430M',\n",
       " 'Deletion',\n",
       " 'Y371H',\n",
       " 'C384R']"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_variation[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2996"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#There are 2996 unique variations\n",
    "len(unique_variation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# remove stopwords and do the stemming\n",
    "corpus = []\n",
    "for i in range(0, 3321):\n",
    "    review = re.sub('[^a-zA-Z]', ' ', training_text['text'][i])\n",
    "    review = review.lower()\n",
    "    review = review.split()\n",
    "    ps = PorterStemmer()\n",
    "    review = [ps.stem(word) for word in review if not word in set(stopwords.words('english'))]\n",
    "    review = ' '.join(review)\n",
    "    corpus.append(review)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['cyclin depend kinas cdk regul varieti fundament cellular process cdk stand one last orphan cdk activ cyclin identifi kinas activ reveal previou work shown cdk silenc increas et v et erythroblastosi viru e oncogen homolog driven activ mapk pathway confer tamoxifen resist breast cancer cell precis mechan cdk modul et activ gener function cdk remain elus demonstr cdk cyclin depend kinas identifi cyclin activ cyclin cyclin orphan cyclin product fam whose mutat caus star syndrom human development anomali whose featur includ toe syndactyli telecanthu anogenit renal malform show star syndrom associ cyclin mutant unabl interact cdk cyclin silenc phenocopi cdk silenc increas c raf confer tamoxifen resist breast cancer cell cdk cyclin phosphoryl et vitro cell posit control et degrad proteasom et protein level increas cell deriv star patient increas attribut decreas cyclin level altogeth result reveal addit regulatori mechan et play key role cancer develop also shed light molecular mechan underli star syndrom cyclin depend kinas cdk play pivot role control number fundament cellular process human genom contain gene encod protein consid member cdk famili owe sequenc similar bona fide cdk known activ cyclin although discov almost ago cdk remain one two cdk without identifi cyclin partner knowledg gap larg imped explor biolog function cdk act posit cell cycl regul cell tumor suppressor other cdk interact et v et erythroblastosi viru e oncogen homolog transcript factor inhibit transcript activ unknown mechan cdk knockdown derepress et increas express c raf protein kinas activ mapk pathway induc resist mcf cell tamoxifen deorphan cdk identifi cyclin product fam bind partner mutat gene predict absenc truncat cyclin associ star syndrom whose featur includ toe syndactyli telecanthu anogenit renal malform heterozyg femal howev function cyclin pathogenesi star syndrom remain unknown show recombin cdk cyclin heterodim activ protein kinas phosphoryl et vitro cyclin silenc phenocopi cdk silenc increas c raf phospho erk express level induc tamoxifen resist estrogen receptor er breast cancer cell show cdk cyclin posit control et degrad proteasom phosphoryl two neighbor serin final detect increas et express level cell deriv star patient demonstr attribut decreas cyclin express level observ cell previou sectionnext sectionresultsa yeast two hybrid h screen unveil interact signal cdk mous protein whose c termin half present strong sequenc homolog human fam gene product whose propos name cyclin thu perform h mate assay determin whether human cdk interact human cyclin fig c longest cdk isoform p express bait protein produc strong interact phenotyp full length cyclin express prey protein detect phenotyp cyclin p cip cdi kap known bind partner cdk fig b cdk cdk also produc h signal cyclin albeit notabl weaker observ cdk fig b interact phenotyp also observ full length cyclin cdk protein express bait prey respect fig test differ isoform cdk cyclin origin altern gene splice two truncat cyclin protein correspond hypothet product two mutat fam gene found star syndrom patient none shorter isoform produc interact phenotyp fig c fig fig new window download pptfig cdk cyclin form interact complex schemat represent differ protein isoform analyz h assay amino acid number indic black box indic intern delet red box indic differ amino acid sequenc compar cdk p b h assay set cdk protein express bait fusion lexa dna bind domain cdk interact protein express prey fusion b transcript activ peg pjg empti bait prey plasmid express lexa b respect lacz use report gene blue yeast indic h interact phenotyp c h assay differ cdk cyclin isoform amino termin region et known interact cdk also assay western blot analysi myc cdk wt kd cycm v express level transfect hek cell e western blot analysi myc cdk wt kd immunoprecipit obtain use anti myc antibodi input correspond g total lysat obtain hek cell coexpress myc cdk wt kd cycm v f western blot analysi immunoprecipit obtain use anti cdk antibodi control goat antibodi human breast cancer mcf cell input correspond g mcf total cell lysat lower band doublet observ upper panel comigr exogen express untag cdk thu correspond endogen cdk upper band doublet correspond nonspecif signal demonstr insensit either overexpress cdk seen left lane silenc cdk fig b anoth experi longer gel migrat shown fig next examin abil cdk cyclin interact express human cell fig e test wild type cdk wt kinas dead kd mutant bear amino acid substitut abolish atp bind express cyclin v myc cdk wt kd human embryon kidney cell line hek express level cyclin v significantli increas upon coexpress myc cdk wt kd lesser extent myc cdk wt kd increas upon coexpress cyclin v fig immunoprecipit myc cdk protein detect presenc cyclin cdk wt kd immunoprecipit protein coexpress pair wise fig e confirm observ detect presenc myc cdk cyclin v immunoprecipit fig b experi confirm lack robust interact cdk p isoform cyclin fig c detect interact endogen protein perform immunoprecipit nontransfect mcf cell deriv human breast cancer cdk cyclin antibodi detect cognat endogen protein western blot readili detect cyclin immunoprecipit obtain cdk antibodi control antibodi fig f result confirm physic interact cdk cyclin human cell unveil hypothes cdk cyclin protein kinas activ produc gst cdk strepii cyclin fusion protein insect cell either individu combin observ gst cdk strepii cyclin copurifi thu confirm interact yet anoth cellular model fig perform vitro kinas assay purifi protein use histon h gener substrat histon h phosphoryl detect lysat cell coexpress gst cdk strepii cyclin phosphoryl detect gst cdk strepii cyclin express alon strepii cyclin coexpress gst cdk kd fig next investig whether et known interact cdk fig c phosphoryl substrat cdk cyclin detect strong phosphoryl et gst cdk strepii cyclin purifi heterodim wherea phosphoryl detect use gst cdk alon gst cdk kd strepii cyclin heterodim fig b fig new window download pptfig cdk cyclin depend protein kinas vitro protein kinas assay histon h lysat insect cell express differ protein purifi glutathion sepharos matrix captur gst cdk wt kd fusion protein alon complex str cycm fusion protein purifi protein express level analyz western blot top upper middl kinas activ determin autoradiographi histon h whose ad amount visual coomassi stain lower middl bottom b use purifi recombin et substrat cdk silenc shown increas et driven c raf transcript activ mapk pathway investig whether cyclin also involv regulatori pathway aim highli specif silenc use sirna pool mix four differ sirna low final concentr nm cdk cyclin sirna pool silenc express cognat target fig c fig interestingli cyclin sirna pool also caus mark decreas cdk protein level fig fig b result shown fig suggest cyclin bind stabil cdk cyclin silenc induc increas c raf protein mrna level fig b c phosphoryl erk erk protein level fig b similarli cdk silenc expect effect cdk cyclin silenc decreas sensit er mcf cell tamoxifen similar extent combin silenc gene result higher resist drug fig c altogeth observ demonstr function interact cyclin cdk neg control et fig new window download pptfig cyclin silenc regul c raf express western blot analysi endogen cdk cyclin express level mcf cell respons sirna mediat gene silenc b western blot analysi endogen c raf express level mcf cell respons cdk cyclin silenc quantif shown fig c quantit rt pcr analysi cdk cyclin c raf mrna level respons cdk upper cyclin lower silenc p p wish explor mechan cdk cyclin control et et short live protein degrad proteasom straightforward hypothesi cdk cyclin posit control et degrad thu examin impact cdk cyclin silenc et express level silenc cdk cyclin caus increas express level exogen express flag et protein fig well endogen et protein fig increas attribut increas et mrna level margin fluctuat respons cdk cyclin silenc fig b examin express level flag tag et protein express alon combin myc cdk cdk kd without cyclin v flag et readili detect express alon lesser extent coexpress cdk kd howev express level dramat decreas coexpress cdk alon cdk cyclin fig b observ suggest endogen cyclin level excess compar cdk mcf cell show major decreas et level observ upon cdk coexpress involv cdk kinas activ treatment cell coexpress flag et cdk cyclin proteasom inhibitor mg larg rescu flag et express level fig b fig new window download pptfig cdk cyclin control et stabil human cancer deriv cell western blot analysi endogen et express level mcf cell respons sirna mediat cdk cyclin silenc quantif shown fig b b western blot analysi exogen express flag et protein level mcf cell cotransfect empti vector coexpress myc cdk wt kd myc cdk cycm v latter cell treat h mg proteasom inhibitor proper express cdk cyclin tag protein verifi western blot analysi c western blot analysi express level exogen express flag et wild type mutant protein mcf cell absenc c respons myc cdk cycm v express quantif shown fig c mass spectrometri analysi recombin et phosphoryl cdk cyclin vitro reveal exist multipl phosphoryl residu among two neighbor phospho serin posit may form phosphodegron fig confirm find compar phosphoryl level recombin et wt et sasa protein mutant bear alanin substitut two serin expect exist multipl phosphoryl site detect small reproduc signific decreas phosphoryl level et sasa compar et wt fig thu confirm ser ser phosphoryl cdk cyclin establish direct link et phosphoryl cdk cyclin degrad examin express level flag et sasa absenc cdk cyclin coexpress differ significantli flag et contrari flag et dbm bear delet n termin destruct box previous shown involv apc cdh mediat degrad et fig c howev contrari flag et wild type express level flag et sasa remain insensit cdk cyclin coexpress fig altogeth result suggest cdk cyclin directli control et degrad phosphoryl two serin final studi lymphoblastoid cell line deriv patient star syndrom bear fam mutat c g predict result aberr splice accord incomplet skew x chromosom inactiv previous found patient detect decreas express level cyclin protein star cell line compar control lymphoblastoid cell line line preced observ detect increas express level et protein star cell line compar control fig fig examin quantit rt pcr mrna express level correspond gene star cell line show decreas express level cyclin mrna express level et mrna similar control cell line fig b demonstr increas et protein express inde result decreas cyclin express observ star patient deriv cell line express cyclin v cell line express caus decreas et protein level fig c fig new window download pptfig decreas cyclin express star patient deriv cell result increas et protein level western blot analysi cyclin et protein level star patient deriv lymphoblastoid cell line control lymphoblastoid cell line deriv healthi individu quantif shown fig b quantit rt pcr analysi cyclin et mrna level cell p c western blot analysi et protein level star patient deriv lymphoblastoid cell line transfect empti vector vector direct express cyclin v anoth western blot reveal endogen exogen express cyclin level shown fig b quantif et protein level shown fig c previou sectionnext sectiondiscussionin work unveil interact cdk last orphan cdk discov pregenom era cyclin cyclin associ human genet diseas far whose function remain unknown closest paralog cdk within cdk famili cdk protein interact l type cyclin interestingli closest paralog cyclin within cyclin famili cyclin fig fact none shorter cdk isoform interact robustli cyclin suggest altern splice cdk gene play import role regul cdk function function relev interact cdk cyclin support differ observ protein seem enhanc stabil judg increas express level partner exogen coexpress fig much reduc endogen cdk express level observ respons cyclin silenc fig fig b cdk subject ubiquitin mediat degrad observ suggest cyclin protect cdk degrad cyclin partner cdk least mcf cell also suggest cyclin stabil enhanc upon bind cdk independ kinas activ seen cyclin c cdk uncov cyclin depend cdk protein kinas activ vitro thu demonstr protein name cdk sole basi amino acid sequenc inde genuin cyclin depend kinas h assay reveal truncat cyclin protein correspond hypothet product two star syndrom associ fam mutat produc interact phenotyp cdk henc regardless whether mutat mrna undergo nonsens mediat decay suggest decreas cyclin mrna level star cell shown fig b give rise truncat cyclin protein femal affect star syndrom must exhibit compromis cdk cyclin kinas activ least tissu specif development stage show et known interactor cdk phosphoryl substrat cdk cyclin vitro cdk cyclin kinas activ posit control et degrad proteasom control seem exert fine mechan judg sensit et level partial decreas cdk cyclin level achiev mcf cell observ star cell respect find offer straightforward explan alreadi report regul et driven transcript c raf respons cdk silenc bring evid cdk cyclin directli control et degrad phosphoryl two neighbor serin may form noncanon trcp phosphodegron dsmcpa none two serin preced prolin conform usual cdk phosphoryl site howev multipl call transcript cdk cdk cdk may belong fig shown phosphoryl varieti motif non prolin direct fashion especi context molecular dock substrat hypothes high affin interact cdk point domain et fig c would allow dock mediat phosphoryl atyp site control et degrad involv number player includ apc cdh cullin ring ligas crl formal identif ubiquitin ligas involv cdk cyclin pathway elucid concert action ubiquitin ligas regul et degrad requir studi result present number signific biolog medic implic first shed light regul et play import role develop frequent deregul mani cancer second result contribut understand molecular mechan caus tamoxifen resist associ reduc cdk express level suggest like cdk cyclin could also predict clinic marker hormon therapi respons er posit breast cancer patient third find offer interest hypothesi molecular mechan underli star syndrom et transgen mice show less twofold overexpress et present sever cranial abnorm observ star patient could thu caus least part increas et protein level anoth expect consequ enhanc et express level would decreas risk develop certain type cancer increas risk develop other studi variou mous model includ model syndrom three copi et exist reveal et dosag repress promot tumor growth henc et exert noncel autonom function cancer intringuingli one star patient identifi far diagnos nephroblastoma final find facilit gener explor biolog function cdk particular role control cell divis previou studi suggest either posit role cell cycl control tumor suppress activ cancer sever growth retard exhibit star patient strongli suggest cdk cyclin play import role control cell prolifer previou sectionnext sectionmateri methodsclon cdk cyclin cdna plasmid construct tamoxifen respons analysi quantit rt pcr mass spectrometri experi antibodi product detail si materi method yeast two hybrid interact assay perform yeast interact mate assay previous describ mammalian cell cultur transfect grew human hek mcf cell dmem supplement vol vol fb invitrogen grew lymphoblastoid cell rpmi glutamax supplement vol vol fb transfect hek mcf cell use lipofectamin invitrogen plasmid lipofectamin rnaimax invitrogen sirna jetprim polyplu plasmid sirna combin accord manufactur instruct transfect lymphoblastoid cell electropor neon invitrogen et stabil studi treat mcf cell h transfect mg fisher scientif h coimmunoprecipit western blot experi collect cell scrape pb centrifug lymphoblastoid cell lyse sonic lysi buffer contain mm glycerophosph mm p nitrophenylphosph mm n morpholino propanesulfon acid mop ph mm egta mm mgcl mm na vanad mm naf mm phenylphosph nonidet p proteas inhibitor mixtur roch spun lysat min g c collect supernat determin protein content use bradford assay perform immunoprecipit experi g total protein lysi buffer preclear lysat l protein g agaros bead incub h c rotat wheel ad g antibodi supernat incub h c rotat wheel ad l protein g agaros bead incub h c rotat wheel collect bead centrifug g c wash three time bead buffer contain mm tri ph mm naf mm nacl mm edta mm egta nonidet p proteas inhibitor coktail roch directli ad sampl buffer wash pellet heat denatur protein ran sampl bi tri sd page transfer protein onto hybond nitrocellulos membran process blot accord standard procedur western blot experi use follow primari antibodi anti myc abcam ab anti v invitrogen r anti tubulin santa cruz biotechnolog b anti cdk covalab pab p santa cruz biotechnolog c anti cycm home made dilut covalab pab p dilut anti raf santa cruz biotechnolog c anti et santa cruz biotechnolog c anti flag sigma f anti actin sigma use hrp coupl anti goat santa cruz biotechnolog sc dilut anti mous bio rad dilut anti rabbit bio rad secondari antibodi reveal blot enhanc chemiluminesc supersign west femto thermo scientif product purif recombin protein gst cdk kd strepii cycm gener recombin bacmid dh bac escherichia coli baculovirus sf cell use bac bac system describ provid invitrogen infect sf cell gst cdk gst cdk kd produc virus coinfect cell strepii cycm produc virus collect cell h infect purifi gst fusion protein spun ml cell resuspend pellet ml lysi buffer pb mm nacl nonidet p mm naf mm glycerophosph mm na vanad contain proteas inhibitor mixtur roch lyse cell sonic spun lysat min g collect solubl fraction ad ml glutathion sepharos matrix incub h c wash four time lysi buffer one time kinas buffer see final resuspend bead l kinas buffer contain vol vol glycerol storag et transform origami de novagen et express vector induc express mm isopropyl thiogalactopyranosid h c purifi et spun ml cell resuspend pellet ml lysi buffer pb mm nacl mm imidazol mm dtt nonidet p contain proteas inhibitor mixtur without edta roch lyse cell bar use cell disruptor spun lysat min g collect solubl fraction ad l cobalt bead thermo scientif h incub c rotat wheel wash four time lysi buffer elut incub bead min elut buffer pb mm imidazol ph contain proteas inhibitor mixtur spun g collect elut protein protein kinas assay mix glutathion sepharos bead harbor gst cdk wt kd either monomer complex strepii cycm bsa mm dtt atp ci atp p histon h et ad kinas buffer mm tri hcl mm mgcl mm egta mm dtt heparin ph total volum l incub reaction min c ad laemli sampl buffer heat denatur sampl ran bi tri sd page cut gel slice detect gst cdk strepii cycm western blot stain gel slice contain substrat coomassi r bio rad dri detect incorpor radioact autoradiographi identifi four unrel girl anogenit renal malform dysmorph facial featur normal intellect syndactyli toe similar combin featur report previous mother daughter pair tabl supplementari note onlin author note clinic overlap town brock syndrom suggest phenotyp repres separ autosom domin entiti mim defin cardin featur syndrom characterist facial appear appar telecanthu broad tripartit nasal tip variabl syndactyli toe hypoplast labia anal atresia urogenit malform fig h also observ varieti featur tabl figur clinic molecular character star syndrom figur clinic molecular character star syndrom f facial appear case appar telecanthu dysplast ear thin upper lip c e toe syndactyli b f case illustr recogniz featur star syndrom specif parent consent obtain public photograph anal atresia hypoplast labia shown g h x ray film feet case show four ray left delta shape th th metatars right h compar clinic pictur array cgh data log ratio repres copi number loss six probe span kb one probe posit within fam delet remov part function gene j schemat structur fam posit mutat fam five code exon box cyclin domain green encod exon horizont arrow indic delet extend case includ exon wherea horizont line exon indic delet found case remov exon sequenc pink horizont bar box indic amplicon use qpcr sequenc one altern exon amplicon indic space constraint mutat dupt case result immedi stop codon g g splice mutat case predict deleteri alter conserv splice donor acceptor site intron respect full size imag kb tabl clinic featur star syndrom case tabl clinic featur star syndrom case full tabl basi phenotyp overlap town brock okihiro feingold syndrom analyz sall ref sall ref mycn found mutat gene supplementari method onlin next carri genom wide high resolut oligonucleotid array compar genom hybrid cgh analysi supplementari method genom dna sever affect individu case lower lid coloboma epilepsi syringomyelia identifi heterozyg delet kb xq remov exon fam fig j use real time pcr confirm delet child exclud unaffect parent supplementari fig onlin supplementari method supplementari tabl onlin cgh custom oligonucleotid array enrich probe xq follow breakpoint clone defin exact delet size bp g del chromosom x ncbi build fig j supplementari fig onlin delet remov code region exon well intron bp bp intron bp sequenc includ utr entir krt p pseudogen ncbi gene id patern proven use routin method find delet overlap fam avail copi number variat cnv databas subsequ carri qpcr analysi three affect individu case mother daughter pair literatur case case detect de novo heterozyg delet kb overlap exon supplementari fig b onlin use xq target array cgh breakpoint clone identifi delet bp g del chromosom x ncbi build fig j supplementari fig remov bp intron exon includ utr bp sequenc found heterozyg fam point mutat remain case fig j supplementari fig supplementari method supplementari tabl case identifi mutat g affect splice donor site intron case identifi frameshift mutat dupt immedi result prematur stop codon n xfsx case detect mutat g alter splice acceptor site intron valid point mutat delet independ round pcr sequenc qpcr confirm patern de novo statu point mutat delet sporad case none mutat seen dna unaffect femal control larger delet involv fam found unrel array cgh investig analyz x chromosom inactiv supplementari method supplementari fig onlin found complet skew x inactiv case almost complet skew case suggest cell carri mutat activ x chromosom growth disadvantag fetal develop use rt pcr rna lymphoblastoid cell case supplementari fig find aberr splice product addit evid mutat allel inactiv furthermor fam subject x inactiv case parent origin delet could determin result lack inform snp case mother case gave birth two boy clinic unaffect sampl avail cannot exclud condit lethal male fetal loss report famili function fam unknown gene consist five code exon bp code region encod protein amino acid genbank list mrna length bp refer sequenc nm express gene est data found adult tissu includ kidney colon cervix uteru heart ncbi express viewer unigen hs express also note list tumor tissu well embryo fetu gene homolog fam ncbi homologen found x chromosom chimpanze dog zebrafish similar gene chromosom howev mous rat true homolog speci similar intronless gene chromosom mous rat like aris retrotransposon insert event murin x chromosom flank gene atp b dusp conserv remnant fam sequenc detect fam contain cyclin box fold domain protein bind domain found cyclin role cell cycl transcript control human phenotyp result cyclin gene mutat yet report homozyg knockout mice ccnd encod cyclin viabl small reduc lifespan also dystroph chang retina like result decreas cell prolifer degener photoreceptor cell embryogenesi cyclin coloc sall nucleu protein cooper mediat transcript repress phenotyp case overlap consider town brock syndrom caus sall mutat carri co immunoprecipit find sall sall would interact fam manner similar observ sall cyclin found fam interact sall sall supplementari fig onlin support hypothesi fam sall particip development pathway fam mutat lead star syndrom growth retard case tabl retin abnorm three case reminisc reduc bodi size retin anomali cyclin knockout mice therefor prolifer defect might partli respons star syndrom address question carri knockdown fam mrna follow prolifer assay transfect hek cell three differ fam specif rnai oligonucleotid result signific reduct fam mrna express prolifer transfect cell supplementari method supplementari fig onlin support link fam cell prolifer found loss function mutat fam result rather homogen clinic phenotyp addit anomali case like result effect kb delet express neighbor gene possibl atp b dusp howev cannot exclud homogen phenotyp result ascertain bia fam mutat includ missens chang could result broader spectrum malform gene caus overlap phenotyp star syndrom town brock syndrom seem act pathway note mycn gene mutat feingold syndrom direct regul cyclin ref thu worth explor whether phenotyp similar feingold star syndrom might explain direct regul fam mycn fam locat approxim mb centromer mecp xq duplic overlap mecp fam describ associ clinic phenotyp femal delet overlap mecp fam observ date although gene fam mecp implic brain develop fam mecp gene region known result x link domin phenotyp thu delet gene allel might lethal male femal',\n",
       " 'abstract background non small cell lung cancer nsclc heterogen group disord number genet proteom alter c cbl e ubiquitin ligas adaptor molecul import normal homeostasi cancer determin genet variat c cbl relationship receptor tyrosin kinas egfr met function nsclc method find use archiv formalin fix paraffin embed ffpe extract genom dna show c cbl mutat occur somat fashion lung cancer c cbl mutat mutual exclus met egfr mutat howev independ p kra mutat normal tumor pairwis analysi signific loss heterozygos loh c cbl locu n none sampl reveal mutat remain copi c cbl c cbl loh also posit correl egfr met mutat observ sampl use select c cbl somat mutat n h q e w obtain caucasian taiwanes african american sampl respect transfect nsclc cell line increas cell viabil cell motil conclus take overal mutat rate c cbl combin somat missens mutat loh clear c cbl highli mutat lung cancer may play essenti role lung tumorigenesi metastasi go introduct us alon year approxim peopl diagnos lung cancer succumb diseas number roughli equival combin mortal rate cancer breast prostat colon liver kidney melanoma addit prognosi usual poor five year surviv rate less also signific ethnic differ lung cancer outcom wors black compar white gender differ also strike women significantli better prognosi compar men number genet alter occur lung cancer exampl nsclc mutat kra p egfr met identifi mani pathway especi receptor tyrosin kinas rtk control c cbl cbl casita b lineag lymphoma mammalian gene locat human chromosom q involv cell signal protein ubiquitin cbl protein belong ring finger class ubiquitin ligas e three homologu c cbl cbl b cbl c cbl cbl b gene ubiquit express highest level hematopoiet tissu c cbl consist four region encod function distinct protein domain n termin tyrosin kinas bind tkb domain linker region catalyt ring finger domain prolin rich region c termin ubiquitin associ uba domain also overlap leucin zipper lz domain tkb ring finger domain essenti ligand induc ubiquitin rtk ring finger domain requir recruit e ubiquitin conjug enzym tkb domain includ four helix bundl h calcium bide ef hand modifi sh domain bind phosphotyrosin residu addit prolin rich region c cbl associ sh domain grb indirectli recruit c cbl rtk via grb adaptor protein c cbl also bind egfr act e target egfr ubiquitin degrad furthermor cbl desensit egf signal oppos cellular prolifer induc egf egf activ also appear activ tyrosin kinas src phosphoryl c cbl turn activ ubiquitin degrad egfr recent studi show defect endocytosi egfr character delet mutant point mutat l r wherebi associ c cbl subsequ ubiquitin impair recent first human c cbl mutat report acut myeloid leukemia aml patient mutat r q inhibit fm like tyrosin kinas flt intern ubiquitin e activ import oncogenesi c cbl dual separ function signal transduct molecul previous shown c cbl import bind crkl bcr abl hematopoiet cell also bind modul function cytoskeleton bind protein like talin paxillin tkb domain import bind number molecul function signal transduct given critic role cbl normal homeostasi cancer hypothes might mutat lung cancer studi report novel c cbl somat mutat n h q e w caucasian taiwanes african american lung cancer patient respect express mutat nsclc cell line lead increas prolifer cell motil show c cbl mutat occur without met egfr mutat mutual exclus loh c cbl locu addit c cbl loh associ either met egfr mutat thu hypothes c cbl mutat might contribut oncogen potenti met egfr lung cancer go method ethic statement written consent research human subject obtain institut review board univers chicago cover research perform laboratori follow contact inform institut review board univers chicago mcgiffert hall woodlawn ave nd floor chicago il written inform consent receiv patient whose tissu sampl use studi tissu sampl lung cancer tissu pair adjac normal lung tissu obtain caucasian african american taiwanes nsclc patient recruit univers chicago hospit chicago usa caucasian african american patient taipei veteran gener hospit taiwan taiwanes patient obtain appropri institut review board permiss inform consent patient sampl men women unknown age diagnosi rang year term tumor type adenocarcinoma squamou cell carcinoma larg cell carcinoma stage stage ii stage iii stage iv tabl cell cultur human non small cell lung carcinoma cell h maintain dmem rpmi respect human embryon kidney cell cultur dmem media supplement fetal bovin serum unit ml penicillin g ml streptomycin invitrogen carlsbad ca cell cultur c humidifi incub contain co c cbl gene mutat analysi exon c cbl gene individu amplifi polymeras chain reaction pcr primer list tabl pcr condit cycl c minut cycl c second c second c minut one cycl c minut pcr product treat exosap usb corpor cleveland oh sequenc big dye termin chemistri appli biosystem foster citi ca sequenc perform forward code strand confirm c cbl alter perform sequenc revers strand well chromatogram analyz mutat use mutat surveyor v softgenet state colleg pa plasmid construct site direct mutagenesi wild type c cbl cdna insert subclon paltermax express vector use xhoi sali restrict enzym site promega madison wi use parent plasmid paltermax c cbl tkb domain doubl mutat n h point mutat q e c termin point mutat w c cbl creat use follow primer gctggcgctaaagaataacccaccttatatcttagac ctaccagatacctaccagtatctccgtactatcttgtc doubl mutat n h ctttacccgactctttgagccctggtcctctttgc q e cagctcctcctttggctgattgtctctggatggtgatc w along complementari primer use quickchang site direct mutagenesi xl kit stratagen la jolla ca accord manufactur instruct construct confirm point mutat standard dna sequenc strand loss heterozygos loh analysi five microsatellit chromosom q within kb downstream c cbl gene control marker p select analysi tabl establish microsatellit marker respect primer sequenc select geneloc databas http genecard weizmann ac il geneloc index shtml weizmann institut scienc rehovot israel primer custom design forward primer fluoresc label end fam pet ned vic appli biosystem primer anneal temperatur duplex score evalu nist primer tool http yellow nist gov dnaanalysi primertoolspag nation institut standard technolog gaithersburg md primer verifi perform pcr control dna isol tk cell resolv product agaros gel band visual uv transillumin genom dna extract tumor sampl pair normal lung tissu primer group multiplex combin shown tabl marker serv intern control check consist pcr peak capillari electrophoresi multiplex pcr carri volum l contain l genom dna ng primer total primer pair dntp x pcr buffer contain mgcl u taq dna polymeras pcr perform abi geneamp pcr system follow condit min c cycl sec c min c min c min c pcr product separ capillari electrophoresi abi xl dna analyz chromatogram analyz peak scanner genemapp softwar appli biosystem allel alter area peak produc dna pcr product quantifi allel ratio allel area calcul tumor pair normal dna sampl qloh allel ratio tumor peak divid allel ratio pair normal sampl c cbl least one q marker least two separ experi sampl consid allel imbal interpret loh sampl evalu least two separ experi sampl show prospect loh c cbl repeat third time includ new control marker bax locu data shown chromosom verifi integr sampl dna transfect c cbl construct cell line transfect use fugen hd roch nutley nj reagent accord manufactur instruct eight g plasmid dna contain either insert empti vector wild type c cbl n h c cbl q e c cbl w cbl use transfect well cultur plate cell harvest h transfect analyz express c cbl knockdown c cbl knockdown perform use lentivir transduct use mission lentivir transduct particl sigma aldrich st loui mo per manufactur instruct briefli h cell well seed well plate infect follow day c cbl lentivir shrna construct gener stabl c cbl knockdown cell line cell select day g ml puromycin c cbl level determin use whole cell lysat immunoblot anti cbl antibodi santa cruz biotechnolog santa cruz ca cell viabil assay cell transfect describ transfect assay forti eight hour transfect viabil cell assess use trypan blue exclus wound heal assay cell seed well plate cultur h confluent medium chang cell transfect describ transfect assay twelv hour transfect straight scratch made across cell layer use ml pipett tip cell gentli wash pb remov cellular debri media replac photograph taken wound region everi h h western blot analysi forti eight hour transfect cell collect wash twice x pb lyse ice cold lysi buffer tri hcl ph nacl deoxychol acid mm edta np mm dtt mm phenylmethylsulfonyl fluorid g ml leupeptin g ml aprotinin minut lysat centrifug rpm minut c protein content supernat measur total cell lysat g well separ sd page electrophoresi gel transfer onto nitrocellulos membran whatman piscataway nj membran block non fat dri milk phosphat buffer salin contain tween pbst x pb tween h room temperatur incub appropri primari antibodi c overnight membran wash three time pbst probe appropri horseradish peroxidas hrp conjug secondari antibodi h room temperatur membran wash three time pbst band visual use western blot chemiluminesc reagent biorad valencia ca chemidoc gel document system biorad valencia ca antibodi obtain santa cruz biotechnolog use follow dilut c cbl c met egfr ubiquitin ha actin flow cytometri cell cycl analysi carri flow cytometri approxim cell grown media contain fb cell harvest trypsin edta treatment wash x pb three time fix ice cold ethanol h cell wash cold pb stain solut contain g ml propidium iodid g ml rnase triton x minut dark cell cycl analysi perform use guava pca flow cytomet guava technolog millipor billerica ubiquitin ligas activ cell maintain cultur dmem supplement fb penicillin unit ml streptomycin g ml transfect g egfr pcdna g ha tag c cbl construct indic use calcium phosphat accord manufactur protocol profect promega madison wi twenti four hour post transfect cell starv overnight dmem supplement fb treat without egf ng ml min cell collect wash two time ice cold pb contain mm sodium orthovanad lyse ice cold lysi buffer mm tri hcl ph mm nacl mm edta triton x glycerol mm sodium orthovanad proteas inhibitor lysat clear debri centrifug g min c egfr immunoprecipit perform g clear lysat use ng rabbit anti egfr protein g plu sepharos overnight c precipit wash time lysi buffer boil laemmli buffer elut immunoblot anti ubiquitin egfr twenti microgram clear lysat immunoblot c cbl construct use anti ha statist analysi mutat rate differ group compar use fisher exact test continu variabl group comparison perform use analysi varianc anova follow sidak adjust multipl comparison experi involv repeat measur time analyz use repeat measur anova greenhous geisser adjust degre freedom analys conduct use stata v softwar stata corpor colleg station tx go result c cbl gene mutat lung cancer investig role c cbl lung cancer analyz genom dna tumor pair normal sampl drawn multipl ethnic lung tumor sampl repres caucasian n african american n taiwanes n lung cancer patient design pair primer sequenc code region c cbl gene span exon tabl identifi uniqu somat mutat c cbl exon among differ patient variat l f known snp rs exon also detect importantli eight novel non synonym mutat confirm sequenc strand c cbl genom dna obtain lung tumor sampl tabl moreov none mutat detect correspond normal tissu indic somat mutat four synonym singl nucleotid variat snv also identifi use studi tabl tabl c cbl mutat analysi lung cancer patient tumor tissu three novel non synonym mutat locat tkb tyrosin kinas bind domain n h q e one ring finger domain v one prolin rich region del atg three c termin region w r k c cbl protein figur figur figur b show model chromatogram repres sampl figur figur c cbl mutat loh non small cell lung cancer q loh c cbl gene pair lung tumor normal lung tissu sampl taiwanes patient n investig loh eight show loh c cbl locu chromosom sampl reveal normal allel contribut microsatellit marker figur c c cbl mutat differ ethnic group c cbl doubl mutant n h found patient overal mutat rate c cbl lung tumor frequenc c cbl mutat highest larg cell carcinoma patient follow squamou carcinoma patient least observ adenocarcinoma ad patient although rate statist signific p mutat rate among caucasian ad sq lc african american ad sq lc ad sq lc taiwanes popul addit two taiwanes patient lung cancer one squamou one adenocarcinoma known snp l f ethnic differ statist signific howev power detect differ low mutat met egfr co associ c cbl alter sinc east asian lung cancer higher frequenc egfr met mutat lung tumor also determin mutat egfr met taiwanes cohort sampl compar result observ c cbl alter loh mutat sampl test find overlap c cbl mutat c cbl loh figur three c cbl mutant includ known l f snp rs one sampl met mutat n egfr mutat l r among sampl loh c cbl locu addit mutat met n egfr exon delet twenti six sampl neither c cbl mutat c cbl loh patient c cbl mutat c cbl loh among sampl met mutat n l w egfr mutat exon delet l r met egfr mutat thu rate met egfr mutat among patient loh c cbl locu similar seen patient without c cbl mutat loh patient p patient identifi mutat c cbl met egfr repres patient analyz taiwanes patient cohort convers taiwanes lung cancer patient identifi mutat either c cbl met egfr combin three gene figur addit determin p kra mutat taiwanes cohort two p kra mutat detect singl kra mutat overlap one p mutat patient also egfr exon delet c cbl mutat p mutat sampl c cbl loh concurr met n mutat thu taiwanes sampl analyz p kra mutat c cbl mutat mutual exclus data shown figur figur c cbl mutat relationship met egfr mutat lung cancer cellular function c cbl alter context lung tumorigenesi e activ intact mutant c cbl protein investig whether differ c cbl mutat affect e activ egfr chosen model substrat c cbl e function c cbl mutant test enhanc ubiquitin activ egfr similar wild type c cbl protein result demonstr catalyt activ c cbl mutant impair egfr substrat figur figur figur ubiquitin viabil express cell cycl analysi variou c cbl mutant b effect lung cancer cell viabil effect repres c cbl mutant three ethnic background lung cancer cell viabil cell line determin n h doubl mutat q e w identifi lung tumor sampl obtain caucasian taiwanes african american respect describ method c cbl wild type wt three mutant express clone paltermax vector cell cell express rel low basal level endogen c cbl data shown transfect effici compar differ group number cell transfect c cbl wild type construct compar control cell transfect empti vector cell transfect n h q e w c cbl mutant construct result increas number viabl cell higher respect rel empti vector control transfect cell significantli differ wild type construct p p p respect figur b rel level c cbl protein whole cell lysat prepar sampl obtain parallel experi determin c cbl protein level sampl repres untransfect empti vector transfect cell compar repres c cbl wt three c cbl mutant compar figur c c effect cell cycl investig increas cell viabil differ c cbl mutant due increas cellular prolifer cell cycl analysi perform cell transfect c cbl wt three differ mutant n h q e w empti vector transfect use control forti eight hour transfect cell cycl analysi perform describ materi method signific chang subg g phase cell cycl among differ mutant compar wt construct p p p respect g phase cell cycl show increas cell number three mutant n h q e w compar wt differ statist signific p figur effect cell motil investig effect express three c cbl mutant cell migrat carri wound heal assay describ materi method close scratch wound monitor h figur sampl repres cell transfect mutant wound gap much smaller seen sampl repres cell transfect c cbl wt p also determin rate wound closur five group h wild type c cbl transfect show open wound n h q e w mutant show open wound respect p figur b figur figur c cbl mutat affect wound heal cell e c cbl knockdown increas cell viabil hypothes loh seen sampl could lead decreas express c cbl thu test effect c cbl knockdown lung cancer cell compar h lung cancer cell express rel high level endogen c cbl data shown c cbl express knock use lentivir construct express c cbl specif shrna compar result transduc scrambl shrna result shown figur identifi sever clone reveal vari degre c cbl knockdown show differ set c cbl lentivir shrna knockdown effici figur clone test clone chosen experi equal amount cell seed well plate cell prolifer measur variou time result depict figur b expect number cell increas time depend fashion rel scrambl shrna control span h p figur b cell cycl phase h cell knock c cbl shrna look compar scrambl shrna discern differ two construct differ phase cell cycl data shown figur figur knockdown c cbl use shrna increas cell prolifer go discuss result demonstr c cbl somat mutat loh lung cancer significantli contribut enhanc cell viabil motil also high preval loh respect c cbl lung tumor harbor met egfr mutat present studi demonstr occurr c cbl mutat lung cancer patient especi differ ancestr variat mutat c cbl recent report juvenil myelomonocyt leukemia myeloid malign aml studi mutat r q locat junction ring finger linker region inhibit fm like tyrosin kinas flt intern ubiquitin thu contribut gain function rtk addit mutat h c r l p map ring finger domain linker region c cbl requir e activ addit homozyg mutat ring finger domain c cbl gene describ result acquir uniparent disomi upd import note result indic loh q locu mutual exclus missens mutat c cbl somat mutat heterozyg mutat aml led abrog e activ lead prolong rtk activ addit mutant locat linker region surround ring finger domain exhibit enhanc akt signal respons cytokin stimul addit shown nh cell neither mutat ring finger linker region caus transform howev certain mutat perturb ubiquitin other affect receptor recycl prolong kinas activ report c cbl mutat map ring finger domain also tkb domain prolin rich domain c termin region none map linker region report aml studi describ addit mutant detect found differ ethnic background exampl n h q e w detect caucasian taiwanes african american respect result point differ lung cancer cancer also genet polymorph among differ race cancer interestingli larg dispar african american ethnic popul lung cancer previous shown low frequenc egfr met mutat african american compar taiwanes caucasian studi number african american sampl analyz rel fewer found mutat uniqu ethnic would behoov us studi genet alter occur determin target therapeut african american result provid evid import c cbl tumorigenesi potenti signal predict base aml data would v ring finger domain mutat would affect e activ also import determin bind partner c cbl tkb domain prolin rich domain mutat previous shown tkb domain bind growth factor receptor import determin cross bind mutant met egfr would also import futur look fluoresc situ hybrid copi number chang c cbl lung cancer c cbl play import role regul rtk mediat signal k poli ubiquitin subsequ downregul rtk follow lysosom degred mono ubiquitin ubiquitin k link chain substrat c cbl may lead enhanc biolog biochem function review hermann et al mutat analyz studi point fact e activ c cbl egfr intact egfr level variou mutant remain figur multipl kinas rtk non rtk could act upon c cbl includ erb pdgfr fm met c kit vegfr flt ron fgfr ir well syk fyn lck fgr lyn c abl lung cancer relev substrat c cbl term degrad signal transduct yet identifi observ c cbl somat mutat especi n h q e w show increas cell viabil cell motil agreement physiolog role cbl regul apoptosi differenti identifi drosophila signific previous shown activ c cbl mutat downregul egfr signal decreas cellular prolifer migrat breast cancer cell line although role c cbl neg regul rtk well substanti therebi suggest natur tumor suppressor studi cancer cell reveal tumor suppressor tumor promot activ depend type c cbl mutat number allel c cbl locu agreement three c cbl mutant describ appear tumor growth metastasi promot properti although mutant outsid ring finger linker region c cbl downstream effect signific caus increas prolifer migrat substrat affect mutat known yet rais possibl cellular function c cbl independ ubiquitin ligas activ area current investig oncogen natur rtk addict cancer growth signal given cluster c cbl egfr met mutat possibl transform effect c cbl mutat like combinatori effect three also show loh c cbl found signific number sampl harbor met egfr mutat fact lung tumor sampl like c cbl mutat addit like harbor c cbl relat loh make c cbl highli mutat molecul lung cancer sinc loh alon enough caus transform event associ mutat met egfr locu yet anoth rtk discuss may play role carcinogenesi predict loh c cbl result haploinsuffici downplay rtk ubiquitin lead hyperact rtk howev whether suffici caus tumorigenesi remain determin consist hypothesi fact c cbl mice increas kinas activ lymphocyt suffici tumor format c cbl loh could also lead increas express c cbl allel compens loss allel altern could form synergi work reduc c cbl level mutat receptor exacerb phenotyp alon previou studi lab other shown east asian lung cancer rel high frequenc gain function mutat rtk egfr met cohort japanes patient activ met mutat identifi splice region delet juxtamembran domain involv e activ c cbl studi also found activ met mutual exclus egfr kra gene mutat fail detect mutat signific number lung tumor sampl obtain african american n caucasian n patient one met mutat identifi group wherea egfr mutat identifi african american caucasian cohort respect egfr mutat earlier identifi one key mutat affect lung adenocarcinoma patient comprehens studi patient studi encompass differ histolog nsclc howev publish seri find mutat c cbl met unlik studi encompass differ subtyp nsclc import note recent shown met mutat lung cancer major germlin report earlier c cbl mutat small cohort taiwanes lung cancer sampl effort understand ethnic differ lung oncogenom also look pax transcript factor pax pax highli express lung cancer howev preferenti express mutat gene lung tumor sampl african american studi show rel high frequenc c cbl mutat lung cancer especi larg cell type among caucasian particularli among african american therefor propos c cbl efficaci target lung cancer african american need substanti import prognosi african american lung cancer especi men much poorer compar caucasian counterpart conclus result present studi demonstr c cbl frequent mutat even lost lung cancer result support role c cbl mutant independ ubiquitin activ given rel high mutat rate c cbl well rtk met egfr like combin effect could synergist promot tumorigenesi']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corpus[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "word_vectorizer = CountVectorizer(ngram_range=(1,1), analyzer='word')\n",
    "sparse_matrix = word_vectorizer.fit_transform(corpus)\n",
    "frequencies = sum(sparse_matrix).toarray()[0]\n",
    "df=pd.DataFrame(frequencies, index=word_vectorizer.get_feature_names(), columns=['frequency'])\n",
    "df.sort_values('frequency', ascending = False, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>mutat</th>\n",
       "      <td>372738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cell</th>\n",
       "      <td>313220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>activ</th>\n",
       "      <td>194042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>express</th>\n",
       "      <td>127028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mutant</th>\n",
       "      <td>126727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>protein</th>\n",
       "      <td>125628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tumor</th>\n",
       "      <td>123192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cancer</th>\n",
       "      <td>121856</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>use</th>\n",
       "      <td>115451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>patient</th>\n",
       "      <td>111606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>figur</th>\n",
       "      <td>109482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fig</th>\n",
       "      <td>107586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>et</th>\n",
       "      <td>107403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>al</th>\n",
       "      <td>105198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <td>102672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>variant</th>\n",
       "      <td>90271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>domain</th>\n",
       "      <td>85137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>type</th>\n",
       "      <td>84286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>function</th>\n",
       "      <td>83547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>brca</th>\n",
       "      <td>82178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>kinas</th>\n",
       "      <td>76192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>result</th>\n",
       "      <td>75456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bind</th>\n",
       "      <td>74959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>studi</th>\n",
       "      <td>74366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>also</th>\n",
       "      <td>66360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sequenc</th>\n",
       "      <td>66167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>data</th>\n",
       "      <td>60541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assay</th>\n",
       "      <td>60247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>analysi</th>\n",
       "      <td>59160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>line</th>\n",
       "      <td>57551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>assess</th>\n",
       "      <td>20114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>respect</th>\n",
       "      <td>20096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>loop</th>\n",
       "      <td>20068</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>import</th>\n",
       "      <td>19978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>thu</th>\n",
       "      <td>19843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mm</th>\n",
       "      <td>19628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>notch</th>\n",
       "      <td>19372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>first</th>\n",
       "      <td>19233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>demonstr</th>\n",
       "      <td>19119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>individu</th>\n",
       "      <td>19111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wherea</th>\n",
       "      <td>19053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissu</th>\n",
       "      <td>18857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>recent</th>\n",
       "      <td>18772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>caus</th>\n",
       "      <td>18749</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>primari</th>\n",
       "      <td>18733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>experi</th>\n",
       "      <td>18669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>imatinib</th>\n",
       "      <td>18586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>erbb</th>\n",
       "      <td>18446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ca</th>\n",
       "      <td>18422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>occur</th>\n",
       "      <td>18353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>erk</th>\n",
       "      <td>18288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>molecular</th>\n",
       "      <td>18285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mek</th>\n",
       "      <td>18281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>possibl</th>\n",
       "      <td>18268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>repres</th>\n",
       "      <td>18093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>small</th>\n",
       "      <td>17985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>time</th>\n",
       "      <td>17981</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confirm</th>\n",
       "      <td>17822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rel</th>\n",
       "      <td>17677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>met</th>\n",
       "      <td>17667</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>200 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           frequency\n",
       "mutat         372738\n",
       "cell          313220\n",
       "activ         194042\n",
       "express       127028\n",
       "mutant        126727\n",
       "protein       125628\n",
       "tumor         123192\n",
       "cancer        121856\n",
       "use           115451\n",
       "patient       111606\n",
       "figur         109482\n",
       "fig           107586\n",
       "et            107403\n",
       "al            105198\n",
       "gene          102672\n",
       "variant        90271\n",
       "domain         85137\n",
       "type           84286\n",
       "function       83547\n",
       "brca           82178\n",
       "kinas          76192\n",
       "result         75456\n",
       "bind           74959\n",
       "studi          74366\n",
       "also           66360\n",
       "sequenc        66167\n",
       "data           60541\n",
       "assay          60247\n",
       "analysi        59160\n",
       "line           57551\n",
       "...              ...\n",
       "assess         20114\n",
       "respect        20096\n",
       "loop           20068\n",
       "import         19978\n",
       "thu            19843\n",
       "mm             19628\n",
       "notch          19372\n",
       "first          19233\n",
       "demonstr       19119\n",
       "individu       19111\n",
       "wherea         19053\n",
       "tissu          18857\n",
       "recent         18772\n",
       "caus           18749\n",
       "primari        18733\n",
       "experi         18669\n",
       "imatinib       18586\n",
       "erbb           18446\n",
       "ca             18422\n",
       "occur          18353\n",
       "erk            18288\n",
       "molecular      18285\n",
       "mek            18281\n",
       "possibl        18268\n",
       "repres         18093\n",
       "small          17985\n",
       "time           17981\n",
       "confirm        17822\n",
       "rel            17677\n",
       "met            17667\n",
       "\n",
       "[200 rows x 1 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[:200]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
